src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -200,6 +200,20 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
     @Param("endDateTime") endDateTime: LocalDateTime? = null,
   ): List<Visit>
 
+  @Query(
+    "SELECT v  FROM Visit v " +
+      "WHERE v.sessionSlot.slotStart >= :startDateTime AND " +
+      "v.prisonerId = :prisonerId AND " +
+      "v.prison.code != :prisonCode AND " +
+      "(cast(:endDateTime as date) is null OR v.sessionSlot.slotEnd < :endDateTime) ORDER BY v.sessionSlot.slotStart,v.id",
+  )
+  fun getVisitsExcludingPrison(
```

Comment:
would actually get rid of the date start and end params and call the method getFutureVisitsExcludingPrison and put logic as follows `v.sessionSlot.slotStart>Now()` you will have to do native also when you do this do the most expensive where last... in other words the date time